### PR TITLE
[DbtProject] RFC: Update API post dogfooding

### PR DESCRIPTION
**Pull request by maximearmstrong**
## Summary & Motivation

This PR updates the API for 2 of the `DbtProject` attributes:
- ~~`target_path` becomes `output_path`~~
    - ~~Target path is the accurate name for the output path in dbt, but this was motivated by the following~~
        - ~~We plan on adding more *Project classes, like `SdfProject`, and would like to standardize the naming across classes~~
        - ~~DbtProject had both the attributes `target` and `target_path`, which means totally different things in this context which is slightly confusing for someone that is not familiar with all dbt concepts.~~
        - Edit: we will keep `target_path` as is to keep the name close to the dbt concept.
- `manifest_preparer` becomes `preparer`
    - This is mostly motivated by #22513, but also
        - we'd like to standardize the naming where possible, and `preparer` would make more sense with other classes like `SdfProject`
        - The `DagsterDbtManifestPreparer` doesn't prepare only the manifest, but also the dependencies with `dbt deps`
            - `DagstetDbtManifestPreparer` and `DbtManifestPreparer` ~~would stay experimental~~ are to be renamed in another PR.
            
Before being merged, these changes should be thoroughly discussed.

## How I Tested These Changes

BK with already existing tests

### CODE REVIEW

**Comment by maximearmstrong** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22534?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#22534** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22534?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#22513** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22513?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @maximearmstrong and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py [hunk 1]
```diff
@@ -125,8 +125,9 @@ class DbtProject(IHaveNew):
     Args:
         project_dir (Union[str, Path]):
             The directory of the dbt project.
-        target_path (Union[str, Path]):
+        output_path (Union[str, Path]):
             The path, relative to the project directory, to output artifacts.
+            In a dbt project, it corresponds to the target path.
```
Comment (sryza) [inline python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py:130]: Given that this class represents a dbt project, I think it's more important to match dbt terminology than to be consistent across different technologies.

File: python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py [hunk 2]
```diff
@@ -180,23 +182,23 @@ def get_env():
...
     project_dir: Path
-    target_path: Path
+    output_path: Path
     target: Optional[str]
     manifest_path: Path
     packaged_project_dir: Optional[Path]
     state_path: Optional[Path]
     has_uninstalled_deps: bool
-    manifest_preparer: DbtManifestPreparer
+    preparer: DbtManifestPreparer
```
Comment (sryza) [inline python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py:191]: This change makes sense to me given that it's doing more than just generating the manifest.

**Review by sryza — CHANGES_REQUESTED** [review]

File: python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py (hunk 1 above)
Comment (maximearmstrong) [reply python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py:130]: Ya, that makes sense. I had mixed feelings about that one.

**Comment by maximearmstrong** [comment]
> I disagree that target_path should be renamed to output_path. If a user is using this integration, we should expect them to be familiar with dbt specific concepts, like "target path" and "target".

Yes, in hindsight that's what makes the most sense.

> I agree with the simplification of manifest_preparer to preparer though. Probably should change to DbtProjectPreparer or something then. I assume we'll introduce a ProjectPreparer base class or something for our future integrations?

Definitely. We are planning on keeping `DagsterDbtManifestPreparer` experimental for now, but `DbtProjectPreparer` is definitely something I had in mind. We plan on doing some user testing, but I think this is a change that should be made beforehand. I'll open a PR.

**Pull request merged**